Cargo.lock

```diff
@@ -1,377 +0,0 @@
-# This file is automatically @generated by Cargo.
```

Comment:
Why does it look like the Cargo.lock is removed in this PR?

Reply:
I think at some point the `Cargo.lock` file was in the git commit index and I later removed it. I took the liberty to add the file to .gitignore.

Reply:
This crate is binary targets, not libraries. Therefore the lockfile is intended to be included. Restore it and take it back out of the .gitignore file. This PR will not be merged otherwise.

Reply:
Please see: https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries